src/drive/ducks/files/Toolbar.jsx

```diff
@@ -24,27 +23,67 @@ import { uploadedFile, downloadFiles, trashFiles } from '../../actions'
 import {
   ShareButton,
   SharedWithMeButton,
+  SharedByMeButton,
   ShareModal,
   SharingDetailsModal
 } from 'sharing'
 import { leave, getSharingDetails } from 'cozy-client'
 
-const { BarRight } = cozy.bar
-
-const toggleShowShareModal = state => ({
-  ...state,
-  showShareModal: !state.showShareModal
-})
-
 const ALERT_LEVEL_INFO = 'info'
 const ALERT_LEVEL_ERROR = 'error'
 const ALERT_LEVEL_SUCCESS = 'success'
 
-class Toolbar extends Component {
+const ActionShare = props => {
+  const { highlighted, shared, onClick, t } = props
+  let label, Component
```

Comment:
Using `let` must warn you that you miss a refactoring somewhere.

Same remark for ugly `if(){} else if {} else`.

Hard to read, hard to understand which value is set. Amha avoinding `if` statement is a good practice.

Reply:
I thought you may say that, but I disagree. I find this entirely readable, way more than any alternative I can think of. Variables and `if` statements are part of the language, it's ok to use them now and then.

Reply:
I hear that:
> Variables and if statements are part of the language, it's ok to use them now and then.

But ;-)

If I take [this very good repo](https://github.com/ryanmcdermott/clean-code-javascript) as an example, I agree with this one

> When you have classes and functions that have if statements, you are telling your user that your function does more than one thing (https://github.com/ryanmcdermott/clean-code-javascript#avoid-conditionals)

For `let`, I don't say you can't use them, I say it's a smell, maybe you should [favor functional over imperative programing](https://github.com/ryanmcdermott/clean-code-javascript#favor-functional-programming-over-imperative-programming) or maybe it's a [side effect](https://github.com/ryanmcdermott/clean-code-javascript#avoid-side-effects-part-1) (with [part2](https://github.com/ryanmcdermott/clean-code-javascript#avoid-side-effects-part-1)), etc.